inventory-repository/src/main/scala/com/normation/inventory/ldap/core/FullInventoryRepositoryImpl.scala

```diff
@@ -237,6 +237,25 @@ class FullInventoryRepositoryImpl(
   }
 
 
+  override def getAllInventories(inventoryStatus : InventoryStatus): Box[Map[NodeId, FullInventory]] = {
+    for {
+      con       <- ldap
+      nodeTree <- con.getTree(inventoryDitService.getDit(inventoryStatus).NODES.dn)
+      nodes     <- sequence(nodeTree.children.values.toSeq) { tree => mapper.nodeFromTree(tree) }
+      machineTree <- con.getTree(inventoryDitService.getDit(inventoryStatus).MACHINES.dn)
+      machines    <- sequence(machineTree.children.values.toSeq) { tree => mapper.machineFromTree(tree) }
+    } yield {
```

Comment:
I think it would be much efficient to get the whole subtree under inventoryDitService.getDit(inventoryStatus), and then special case by its machine/node chilldren, or even get the children.children and pattern match on dn and return a Box[(nodes, machines)]